PersianCalendar@oxygenws.com/PersianDate.js

```diff
@@ -1,104 +1,92 @@
+
+
 /*
- * Based on a code from http://farhadi.ir
+ * Edit by: jdf.scr.ir
  */
 
 var PersianDate = {
-    g_days_in_month: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
-    p_days_in_month: [31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29],
+    // g_days_in_month: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+    // p_days_in_month: [31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29],
     p_month_names: ['فروردین', 'اردیبهشت', 'خرداد', 'تیر', 'مرداد', 'شهریور', 'مهر', 'آبان', 'آذر', 'دی', 'بهمن', 'اسفند']
 };
 
-PersianDate.persianToGregorian = function (py, pm, pd) {
-    py = parseInt(py) - 979;
-    pm = parseInt(pm) - 1;
-    pd = parseInt(pd) - 1;
-
-    let p_day_no = 365 * py + parseInt(py / 33) * 8 + parseInt((py % 33 + 3) / 4);
-    for (let i = 0; i < pm; i++) {
-        p_day_no += PersianDate.p_days_in_month[i];
-    }
-
-    p_day_no += pd;
-
-    let g_day_no = p_day_no + 79;
-
-    let gy = 1600 + 400 * parseInt(g_day_no / 146097);
-    /* 146097 = 365*400 + 400/4 - 400/100 + 400/400 */
-    g_day_no %= 146097;
-
-    let leap = true;
-    /* 36525 = 365*100 + 100/4 */
-    if (g_day_no >= 36525) {
-        g_day_no--;
-        gy += 100 * parseInt(g_day_no / 36524);
-        /* 36524 = 365*100 + 100/4 - 100/100 */
-        g_day_no %= 36524;
-
-        if (g_day_no >= 365) {
-            g_day_no++;
-        } else {
-            leap = false;
-        }
-    }
-
-    gy += 4 * parseInt(g_day_no / 1461);
-    /* 1461 = 365*4 + 4/4 */
-    g_day_no %= 1461;
-
-    if (g_day_no >= 366) {
-        leap = false;
-
-        g_day_no--;
-        gy += parseInt(g_day_no / 365);
-        g_day_no %= 365;
-    }
+PersianDate.checkDate = function (py, pm, pd) {
+    return !(py < 0 || py > 32767 || pm < 1 || pm > 12 || pd < 1 || pd > PersianDate.pDaysInMonth(py, pm));
+};
 
-    let i = 0;
-    for (; g_day_no >= PersianDate.g_days_in_month[i] + (i === 1 && leap); i++) {
-        g_day_no -= PersianDate.g_days_in_month[i] + (i === 1 && leap);
+PersianDate.isKabiseh = function (py, outType = 'bool') {
```

Comment:
Use international variable names. (isLeap)
And... I don't like this function. "isSomething" should return a boolean!